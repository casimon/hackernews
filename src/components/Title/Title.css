.Title {
  padding: 10px;
}

.Title-link {
  color: #fb651e;
  font-size: 18px;
  text-decoration: none;
}
