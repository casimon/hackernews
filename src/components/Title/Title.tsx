import React from 'react'
import './Title.css'

const Title = ({ score, title, url }: any) => {
  return (
    <div className='Title'>
      <a
        href={url}
        className='Title-link'
        target='_blank'
        rel='noopener noreferrer'
      >
        {title} {score && `(Score - ${score})`}
      </a>
    </div>
  )
}

export default Title
