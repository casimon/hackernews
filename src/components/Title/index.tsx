import React from 'react'
import './Title.css'

const Title = ({ score, title, url, by, time }: any) => {
  return (
    <div className='Title'>
      <a
        href={url}
        className='Title-link'
        target='_blank'
        rel='noopener noreferrer'
      >
        {title}
      </a>
      <br />
      <span>{score && `Score - ${score}, By ${by}`}</span>
    </div>
  )
}

export default Title
