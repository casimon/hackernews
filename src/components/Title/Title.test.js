import React from 'react'

import { render } from '@testing-library/react'
import Title from './Title'

const init = (props) => render(<Title {...props} />)

describe('Test title component', () => {
  test('should render title with no props', () => {
    const { container } = init()
    expect(container).toMatchSnapshot()
  })

  test('should render title with props', () => {
    const { container } = init({
      title: 'Test',
      score: 1,
      url: 'http://www.test.com',
      by: 'Simon Gamble',
    })
    expect(container).toMatchSnapshot()
  })
})
