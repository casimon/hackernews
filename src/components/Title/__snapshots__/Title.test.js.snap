// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test title component should render title with no props 1`] = `
<div>
  <div
    class="Title"
  >
    <a
      class="Title-link"
      rel="noopener noreferrer"
      target="_blank"
    />
    <br />
    <span />
  </div>
</div>
`;

exports[`Test title component should render title with props 1`] = `
<div>
  <div
    class="Title"
  >
    <a
      class="Title-link"
      href="http://www.test.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Test
    </a>
    <br />
    <span>
      Score - 1, By Simon Gamble
    </span>
  </div>
</div>
`;
